interp/interp.go

```diff
@@ -84,8 +87,10 @@ func (pl *PiecewiseLinear) Fit(xs, ys []float64) error {
 		}
 		pl.slopes[i] = (ys[i+1] - ys[i]) / dx
 	}
-	pl.xs = xs
-	pl.ys = ys
+	pl.xs = make([]float64, n)
+	pl.ys = make([]float64, n)
+	copy(pl.xs, xs)
+	copy(pl.ys, ys)
```

Comment:
It can lead to hard to detect bugs if xs or ys is modified after falling Fit.

Reply:
Can you elaborate? I don't think we've ever considered this a reason to allocate, but maybe there is a first time.

Reply:
I mean sth like this:

```
xs = make([]float64, 10)
ys = make([]float64, 10)
// set values
fitter1.Fit(xs, ys)
ys[0] = -10
fitter2.Fit(xs, ys)
// fitter1 is now broken because we made a shallow copy
```

Reply:
Yeah, understood. Either we document that a copy is made, or we document that the receiver takes ownership of the slices that it is given and they should not be altered after they have been passed to it. Both of these approaches are used.

Reply:
I'm inclined towards the no-copy, but it's worth waiting for other opinions.

Reply:
If the dominant style in `gonum` is to avoid defensive copying, then I'll revert this.

Reply:
Yeah, it varies. I don't use this kind of thing often enough to have a strongly defensible opinion, so I would wait for one of the others.

Reply:
I'm also slightly inclined towards the no-copy. If the user modifies the slices, they're probably not interested in the interpolation or should not use it as a backup storage. It could even hide their bugs because with a copy the interpolator would keep returning good looking results, with the no-copy something noticeably bad would happen.

Reply:
I think we should copy. IMO it's generally good practice to do defensive copying when passing into types that have state (or retrieving from them). It's so easy to cause bugs you don't expect. When you call a constructor, it "feels like" the type now just exists and is immutable unless you call one of the methods. IMO things should act like that unless there's a really good reason not to.

Reply:
Ok, fine for me.

Reply:
I'll leave it as it is then.